sycl/source/detail/scheduler/graph_builder.cpp

```diff
@@ -561,11 +593,19 @@ AllocaCommandBase *Scheduler::GraphBuilder::getOrCreateAllocaForReq(
 
   if (!AllocaCmd) {
     if (!ForceFullReq && IsSuitableSubReq(Req)) {
-      if (Req->MDims != 1)
-        throw runtime_error("OpenCL only supports 1D sub buffers");
-
-      auto *ParentAlloca = getOrCreateAllocaForReq(Record, Req, Queue, true);
+      // Get parent requirement. It's hard to get right parents' range
```

Comment:
Do we still need ForceFullReq flag?

Reply:
I don't see any reason to stick with this flag. Will remove it with another patch.